### PR TITLE
add dock

### PR DESCRIPTION
**Pull request by FindHao**
update package.py to include variants with optimizations. We add optional jemalloc as the optimization, and add the support for arm and nvhpc compilers.

### CODE REVIEW

File: var/spack/repos/builtin/packages/dock/package.py [hunk 1]
```diff
@@ -19,12 +19,15 @@ class Dock(Package):
     url      = "file://{0}/dock.6.9_source.tar.gz".format(os.getcwd())
     manual_download = True
 
-    version('6.9', sha256='c2caef9b4bb47bb0cb437f6dc21f4c605fd3d0d9cc817fa13748c050dc87a5a8')
+    version('6.9', preferred=True, sha256='c2caef9b4bb47bb0cb437f6dc21f4c605fd3d0d9cc817fa13748c050dc87a5a8')
+    version('opt', git='file://{0}/dock'.format(os.getcwd()))
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/dock/package.py:23]: What would be the purpose of this version?

**Review by alalazo — CHANGES_REQUESTED** [review]

**Comment by OliverPerks** [comment]
It might be worth holding off on this.
Looking at the report from the hackathon we need to add the `arm` and `nvhpc` patch files, and the compiler config files.

As for the `jemalloc` feature, this shows a good performance gain, but again we would want to make sure we are adding it properly (rather than hard-coding it in the config files).
I'm happy to work with you to add this all in a more Spack friendly way.

**Comment by adamjstewart** [comment]
> It might be worth holding off on this.

Converted to a draft, feel free to mark the PR as "Ready for review" when it's ready.

**Comment by alalazo** [comment]
Closing, since there has been no activity in a while. Feel free to reopen if you want to continue work on this.

**Pull request closed without merging**